silhouette/test/com/mohiva/play/silhouette/impl/providers/OAuth1ProviderSpec.scala

```diff
@@ -78,6 +79,19 @@ abstract class OAuth1ProviderSpec extends SocialProviderSpec[OAuth1Info] {
           redirectLocation(result) must beSome.which(_ == c.oAuthSettings.authorizationURL)
       }
     }
+    "resolves relative redirectURLs  before starting the flow" in new WithApplication {
+      implicit val req = FakeRequest(GET, "/").withHeaders(HeaderNames.HOST -> "www.example.com")
+      c.oAuthSettings.callbackURL returns "/callback-url"
+      val resolvedCallbackURL = "http://www.example.com/callback-url"
+
+      c.oAuthService.retrieveRequestToken(resolvedCallbackURL) returns Future.successful(c.oAuthInfo)
+      c.oAuthService.redirectUrl(c.oAuthInfo.token) returns c.oAuthSettings.authorizationURL
+      c.oAuthTokenSecretProvider.build(any)(any) returns Future.successful(c.oAuthTokenSecret)
+      c.oAuthTokenSecretProvider.publish(any, any)(any) returns Results.Redirect(c.oAuthSettings.authorizationURL)
+
+      await(c.provider.authenticate())
+      there was one(c.oAuthService).retrieveRequestToken(resolvedCallbackURL)
+    }
 
```

Comment:
You should also test the HTTPS version. The HTTPS test could also be tested without the leading slash to see if a slash will be added to the URL.